python/ray/data/random_access_dataset.py

```diff
@@ -0,0 +1,265 @@
+import bisect
+import logging
+import random
+import time
+from collections import defaultdict
+from typing import List, Any, Generic, Optional, TYPE_CHECKING
+
+import ray
+from ray.types import ObjectRef
+from ray.data.block import T, BlockAccessor
+from ray.data.impl.remote_fn import cached_remote_fn
+
+if TYPE_CHECKING:
+    from ray.data import Dataset
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
This is not a subclass of Dataset (probably doesn't make sense to be), should it be modeled as RandomAccessor or DatasetRandomAccessor?

Reply:
That's true, though it doesn't seem to be as nice of a name.